BuilderBuilder.Test/AssertHelper.cs

```diff
@@ -1,86 +1,73 @@
-        ) {
-            var fieldsWithInverseType = fields.Select(t => (t.type, t.name, Field.InverseHandlingType.None)).ToArray();
-            AssertBuilderEntity(entityResult, name, persistable, fieldsWithInverseType);
-        }
+    public static void AssertBuilderEntity(
+        BuilderEntity entityResult, string name, bool persistable, params (string type, string name)[] fields
+    ) {
+        var fieldsWithInverseType = fields.Select(t => (t.type, t.name, Field.InverseHandlingType.None)).ToArray();
+        AssertBuilderEntity(entityResult, name, persistable, fieldsWithInverseType);
+    }
 
-        public static void AssertBuilderEntity(BuilderEntity entityResult, string name, bool persistable,
-            params (string type, string name, Field.InverseHandlingType inverse)[] fields
-        ) {
-            Assert.AreEqual(name, entityResult.Name);
+    public static void AssertBuilderEntity(BuilderEntity entityResult, string name, bool persistable,
+        params (string type, string name, Field.InverseHandlingType inverse)[] fields
+    ) {
+        Assert.Equal(name, entityResult.Name);
 
-            Assert.AreEqual(persistable, entityResult.Persistable);
+        Assert.Equal(persistable, entityResult.Persistable);
 
-            var expectedFields = fields.Select(t => new Field(t.type, t.name, t.inverse));
+        var expectedFields = fields.Select(t => new Field(t.type, t.name, t.inverse));
 
-            AssertHelper.ListEq(entityResult.Fields, f => $"{f.Type} {f.Name} {f.InverseHandling}", expectedFields);
-        }
+        ListEq(entityResult.Fields, f => $"{f.Type} {f.Name} {f.InverseHandling}", expectedFields);
+    }
 
-        public static void ListEq<T, C>(IEnumerable<T> result, Func<T, C> compareBy, params T[] expected) {
-            ListEq(result, compareBy, expected.AsEnumerable());
-        }
-        public static void ListEq<T, C>(IEnumerable<T> result, Func<T, C> compareBy, params C[] expected) {
-            ListEq(result, compareBy, expected.AsEnumerable());
-        }
-        public static void ListEq<T>(IEnumerable<T> result, params T[] expected) {
-            ListEq(result, expected.AsEnumerable());
-        }
+    private static void ListEq<T, TC>(IEnumerable<T> result, Func<T, TC> compareBy, IEnumerable<T> expected) {
+        ListEq(result.Select(compareBy), expected.Select(compareBy));
+    }
 
-        public static void ListEq<T, C>(IEnumerable<T> result, Func<T, C> compareBy, IEnumerable<T> expected) {
-            ListEq(result.Select(compareBy), expected.Select(compareBy));
-        }
-        public static void ListEq<T, C>(IEnumerable<T> result, Func<T, C> compareBy, IEnumerable<C> expected) {
-            ListEq(result.Select(compareBy), expected);
+    private static void ListEq<T>(IEnumerable<T> result, IEnumerable<T> expected) {
+        var actualArr = result as T[] ?? result.ToArray();
+        var expectedArr = expected as T[] ?? expected.ToArray();
```

Comment:
The `ToArray` makes sure the enumerable is only enumerated once, since the parameter is used multiple times in this method. Usually multiple enumerations shouldn't change the result of the enumeration, but I've seen cases where it did so I make sure they are always enumerated once. 

The `as T[]` looks like a micro-optimization that I will remove.